### PR TITLE
Front end sign up

### PR DESCRIPTION
**Pull request by greenc123**
* Signin links to signup page
* Adds integration tests

TODO:
* Needs to send email

### CODE REVIEW

File: views/signup.ejs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    fetch('/api/users', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        username: username.value,
+        email: email.value
+      })
+    }).then(r => r.json()).then((r) => {
```
Comment (AliceHuang1027) [inline views/signup.ejs:49]: （r）can just be written as () since r is not used...

File: views/signup.ejs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/open-fonts@1.1.1/fonts/inter.min.css">
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@exampledev/new.css@1.1.2/new.min.css">
```
Comment (AliceHuang1027) [inline views/signup.ejs:2]: i wonder if we need to adjust the margin for some of the div, input and button...

File: views/signup.ejs (hunk 1 above)
Comment (greenc123) [reply views/signup.ejs:49]: I'm going to send an email with the response

File: views/signup.ejs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+<h1>Learn Databases</h1>
+
+<p>Create Account</p>
+
+<div>
+  <input
+    class="square email"
+    type="text"
```
[the remark this answers is not in the document]
Comment (greenc123) [reply views/signup.ejs:11]: include what?

File: views/signup.ejs (hunk 2 above)
Comment (greenc123) [reply views/signup.ejs:2]: maybe in the future but for right now the site doesn't have to look perfect

File: views/signup.ejs (hunk 3 above)
[the remark this answers is not in the document]
Comment (greenc123) [reply views/signup.ejs:11]: changed type to email

File: views/signup.ejs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/open-fonts@1.1.1/fonts/inter.min.css">
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@exampledev/new.css@1.1.2/new.min.css">
+
+<h1>Learn Databases</h1>
+
+<p>Create Account</p>
+
+<div>
+  <input
+    class="square email"
```
[the remark this answers is not in the document]
Comment (greenc123) [reply views/signup.ejs:10]: It was part of the old css

**Comment by kimong1** [comment]
One thing needs to be added in sign up page. When users click `Create Account`, it redirects to sign in page immediately. I think it should show some message to tell users know what to do. For example:
```
submit.addEventListener('click', () => {
    ... ...
  alert("Check your email and set your password")
  return location.href = '/signin'
```

File: views/signup.ejs (hunk 1 above)
Comment (songz) [reply views/signup.ejs:69]: @greenc123 this actually redirects. This code here is the front end.

File: views/signup.ejs (hunk 2 above)
Comment (songz) [reply views/signup.ejs:2]: Will resolve in this issue: https://github.com/garageScript/databases/issues/110

**Pull request merged**